modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1648,21 +1649,51 @@ def fold(self, axis, func):
             self._column_widths,
         )
 
-    def infer_types(self, columns_list: List[str]) -> "PandasDataframe":
+    def infer_objects(self) -> "PandasDataframe":
+        """
+        Attempt to infer better dtypes for object columns.
+
+        Attempts soft conversion of object-dtyped columns, leaving non-object and unconvertible
+        columns unchanged. The inference rules are the same as during normal Series/DataFrame
+        construction.
+
+        Returns
+        -------
+        PandasDataframe
+            A new PandasDataframe with the inferred schema.
+        """
+        obj_cols = [
+            col for col, dtype in enumerate(self.dtypes) if is_object_dtype(dtype)
+        ]
+        return self.infer_types(obj_cols)
+
+    def infer_types(self, col_labels: List[str]) -> "PandasDataframe":
         """
         Determine the compatible type shared by all values in the specified columns, and coerce them to that type.
 
         Parameters
         ----------
-        columns_list : list
+        col_labels : list
             List of column labels to infer and induce types over.
 
         Returns
         -------
         PandasDataframe
             A new PandasDataframe with the inferred schema.
         """
-        pass
+        # Compute dtypes on the specified columns, and then set those dtypes on a new frame
+        new_cols = self.take_2d_labels_or_positional(col_labels=col_labels)
+        new_cols_dtypes = new_cols.tree_reduce(0, pandas.DataFrame.infer_objects).dtypes
+        new_dtypes = self.dtypes.copy()
```

Comment:
do we have to do a copy here or can we just update the dtypes in place with the same reference?

Reply:
Yes, or else `infer_types` would mutate the dtypes of the original dataframe.

```
>>> df = pd.DataFrame({"a": ["s", 2, 3]}).iloc[1:]
>>> df.dtypes
a    object
dtype: object
>>> df.infer_objects()
<modin.core.storage_formats.pandas.query_compiler.PandasQueryCompiler object at 0x7fb0124b7880>
>>> df.dtypes
a    int64
dtype: object
```